### PR TITLE
Move dependency versions to a machine-updatable TOML catalog

### PR DESCRIPTION
**Pull request by PawelLipski**
Based on #939.

Gradle is love, Gradle is life 🥰

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -117,17 +117,23 @@ To watch the logs of this IntelliJ instance, run `tail -f build/idea-sandbox/sys
...
 See [Robot plugin](https://github.com/JetBrains/intellij-ui-test-robot)
 and [a preso on testing UI of IntelliJ Plugins](https://slides.com/plipski/intellij-plugin-ui-testing) for more details.
 
 
-## Check dependency updates
+## Update version catalog
 
-`./gradlew dependencyUpdates`
+```shell
+./gradlew versionCatalogUpdate
```
Comment (PawelLipski) [inline CONTRIBUTING.md:131]: This updates `gradle/libs.versions.toml`.

`dependencyUpdates` only printed out the new versions, to be manually updated ☹️

File: build.gradle [hunk 2]
```diff
@@ -1,6 +1,6 @@
 buildscript {
   dependencies {
-    classpath group: 'org.jsoup', name: 'jsoup', version: '1.15.1'
+    classpath libs.jsoup
```
Comment (PawelLipski) [inline build.gradle:3]: Identifiers autogenerated based on `gradle/libs.versions.toml`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -93,9 +93,6 @@ To overwrite them, use `compileJavaJvmArgs` Gradle project property
 (e.g. `./gradlew -PcompileJavaJvmArgs='-Xmx2g -XX:+HeapDumpOnOutOfMemoryError' build`,
 or equivalently with an env var: `ORG_GRADLE_PROJECT_compileJavaJvmArgs='-Xmx2g -XX:+HeapDumpOnOutOfMemoryError' ./gradlew build`).
 
-By default, Lombok's annotation processor runs on the fly and Delomboked sources are **not** saved to <subproject>/build/delombok/...<br/>
-To enable Delombok, set `useDelombok` Gradle project property (e.g. `./gradlew -PuseDelombok build`).
```
Comment (PawelLipski) [inline CONTRIBUTING.md:97]: That didn't even work anymore, so probably nobody used that in a long time :trollface:

File: build.gradle [hunk 4]
```diff
@@ -9,19 +9,16 @@ plugins {
...
-  id 'org.ajoberstar.grgit' version '5.0.0' apply false
-  id 'org.checkerframework' version '0.6.11' apply false
-  id 'org.jetbrains.grammarkit' version '2021.2.2' apply false
-  id 'org.jetbrains.intellij' version '1.6.0' apply false
-  // As per https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library:
-  //   If a plugin supports multiple platform versions, it must (...) target the lowest bundled stdlib version
-  // which is 1.4.0 in our case (as this is the Kotlin version bundled into IntelliJ 2020.3).
-  id 'org.jetbrains.kotlin.jvm' version '1.4.0' apply false
+  alias libs.plugins.versions
+  alias libs.plugins.versionCatalogUpdate
```
Comment (PawelLipski) [inline build.gradle:13]: Added

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    "ideProbe-junitDriver",
+    "ideProbe-robotDriver",
+]
+powerMock = [
+    "powerMock-apiMockito2",
+    "powerMock-moduleJunit4",
+]
+
+[plugins]
+# @pin because 6.5.x leads to a build error: https://github.com/freefair/gradle-plugins/issues/554
```
Comment (PawelLipski) [inline gradle/libs.versions.toml:38]: `@pin`, as in `./gradlew versionCatalogUpdate` should NOT update this one

**Review by mkondratek — APPROVED** [review]

**Pull request merged**